scripts/hud/jumpstats.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+
+class JumpStats {
+	/** @type {Label} @static */
+	static label = $('#JumpStatsLabel');
+	/** @type {Panel} @static */
+	static container = $('#JumpStatsContainer');
+	/** @type {Panel} @static */
+	static panel = $.GetContextPanel();
+
+	static bufferLength = 10;
+
+	static onJumpEnded() {
+		const lastJumpStats = MomentumMovementAPI.GetLastJumpStats();
+		if (lastJumpStats.jumpCount < this.jumpStatsConfig.statsFirstPrint) {
+			return;
+		}
+
+		if (
+			(lastJumpStats.jumpCount - this.jumpStatsConfig.statsFirstPrint) % this.jumpStatsConfig.statsInterval !==
+			0
+		) {
+			return;
+		}
+
+		this.addToBuffer(this.countBuffer, lastJumpStats.jumpCount + ':');
+		this.addToBuffer(this.takeoffSpeedBuffer, lastJumpStats.takeoffSpeed.toFixed());
+		this.addToBuffer(this.speedDeltaBuffer, lastJumpStats.jumpSpeedDelta.toFixed());
+		this.addToBuffer(this.takeoffTimeBuffer, this.makeTime(lastJumpStats.takeoffTime.toFixed(3)));
+		this.addToBuffer(this.timeDeltaBuffer, lastJumpStats.timeDelta.toFixed(3));
+		this.addToBuffer(this.strafesBuffer, lastJumpStats.strafeCount);
+		this.addToBuffer(this.syncBuffer, lastJumpStats.strafeSync.toFixed(2));
+		this.addToBuffer(this.gainBuffer, lastJumpStats.speedGain.toFixed(2));
+		this.addToBuffer(this.strafeRatioBuffer, lastJumpStats.strafeRatio.toFixed(2));
+		this.addToBuffer(this.heightDeltaBuffer, lastJumpStats.heightDelta.toFixed(1));
+		this.addToBuffer(this.efficiencyBuffer, lastJumpStats.jumpEfficiency.toFixed(2));
+
+		this.setText();
+	}
+
+	static initializeBuffer(size) {
+		return Array(size).fill('');
+	}
+
+	static addToBuffer(buffer, value) {
+		buffer[buffer.length - 1] += '\n';
```

Comment:
I think `Array.at()` only works to fetch the value in the array at the desired position. Since the purpose here is to assign a value to the last position in the array, something else will be needed. I think python supports negative indices for accessing from the end of an array, but js seems to only allow `array[array.length - 1]`.

Reply:
![image](https://user-images.githubusercontent.com/22461508/205453290-ba5ccbbd-6fa0-478f-ac9d-8d11caa2a78a.png)

Reply:
Oh my bad, you're completely right!